### PR TITLE
Run SDK lint jobs

### LINKED ISSUE #6628: [sdk/python] `make lint` not running in CI?
Running `make lint` locally in `sdk/python` was resulting in:

```
LINT:
pipenv run mypy ./lib/pulumi --config-file=mypy.ini
lib/pulumi/x/automation/_minimum_version.py:15: error: Skipping analyzing 'semver': found module but no type hints or library stubs
lib/pulumi/x/automation/_cmd.py:62: error: Item "None" of "Optional[IO[bytes]]" has no attribute "readline"
lib/pulumi/runtime/rpc.py:118: error: Name 'CustomResource' is not defined
lib/pulumi/runtime/resource.py:524: error: Incompatible types in assignment (expression has type "Optional[str]", variable has type "str")
lib/pulumi/runtime/resource.py:526: error: Incompatible types in assignment (expression has type "Optional[Dict[str, PropertyDependencies]]", variable has type "Dict[str, PropertyDependencies]")
lib/pulumi/output.py:393: error: Name 'all_outputs' already defined on line 391
lib/pulumi/output.py:400: error: Argument 1 to "get_resources" has incompatible type "object"; expected "List[Any]"
lib/pulumi/output.py:401: error: Argument 1 to "is_known" has incompatible type "object"; expected "List[Any]"
lib/pulumi/output.py:402: error: Argument 1 to "is_secret" has incompatible type "object"; expected "List[Any]"
lib/pulumi/x/automation/_local_workspace.py:20: error: Skipping analyzing 'semver': found module but no type hints or library stubs
lib/pulumi/x/automation/_local_workspace.py:20: note: See https://mypy.readthedocs.io/en/latest/running_mypy.html#missing-imports
Found 10 errors in 6 files (checked 58 source files)
make: *** [lint] Error 1
```

https://github.com/pulumi/pulumi/pull/6627 fixes these issues.

Are we not running the lint target in CI?

### PR DESCRIPTION
**Pull request by pgavlin**
Just what it says on the tin. Fixes #6628.

### CODE REVIEW

File: sdk/python/lib/pulumi/output.py [hunk 1]
```diff
@@ -256,7 +256,7 @@ def from_input(val: Input[T]) -> 'Output[T]':
             # Since Output.all works on lists early, serialize the class's __dict__ into a list of lists first.
             # Once we have a output of the list of properties, we can use an apply to re-hydrate it back as an instance.
             items = [[k, Output.from_input(v)] for k, v in val.__dict__.items()]
-            fn = cast(Callable[[List[Any]], T], lambda props: typ(**{k: v for k, v in props}))  # type: ignore
+            fn = cast(Callable[[List[Any]], T], lambda props: typ(**props))  # type: ignore
```
Comment (komalali) [inline sdk/python/lib/pulumi/output.py:259]: This change breaks the `Output.from_input` tests. https://github.com/pulumi/pulumi/pull/6674/checks?check_run_id=2241937838#step:17:2096

Might need to add `# pylint: disable=unnecessary-comprehension` like line 268

**Comment by komalali** [comment]
Ugh sorry I accidentally hit close when I was trying to comment that the break you're seeing now is my fault. https://github.com/pulumi/pulumi/pull/6677

File: sdk/python/lib/pulumi/output.py (hunk 1 above)
Comment (pgavlin) [reply sdk/python/lib/pulumi/output.py]: Done.

**Review by justinvp — APPROVED** [review]
LGTM

**Review by komalali — APPROVED** [review]

**Pull request merged**